### PR TITLE
UI: Client Count remove client-side namespace filtering

### PR DESCRIPTION
**Pull request by hashishaw**
### Description
This sidebranch PR removes the client-side namespace filtering, and updates the client-side mount filtering to be consistent. I also removed the average clients stat block.

image

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: ui/app/routes/vault/cluster/clients/counts.ts [hunk 1]
```diff
@@ -116,7 +116,11 @@ export default class ClientsCountsRoute extends Route {
       activityError,
       config,
       // activity.startTime corresponds to first month with data, but we want first month returned or requested
-      startTimestamp: this.paramOrResponseTimestamp(params?.start_time, activity?.byMonth[0]?.timestamp),
+      // unless no months present, then we can fallback to response's start time
+      startTimestamp: this.paramOrResponseTimestamp(
```
Comment (hashishaw) [inline ui/app/routes/vault/cluster/clients/counts.ts:120]: Fixes issue where dates are not set on load for enterprise when no data is available

File: ui/tests/acceptance/clients/counts/acme-test.js [hunk 2]
```diff
@@ -85,13 +87,11 @@ module('Acceptance | clients | counts | acme', function (hooks) {
         `${formatNumber([nsTotals.acme_clients])}`,
         'renders total acme clients for namespace'
       );
-    // there is only one month in the stubbed data, so in this case the average is the same as the total new clients
+
+    // TODO: update this
```
Comment (hashishaw) [inline ui/tests/acceptance/clients/counts/acme-test.js:91]: This is still a task -- I need to update the average calculation

File: ui/tests/acceptance/clients/counts/overview-test.js [hunk 3]
```diff
@@ -9,28 +9,32 @@ import { setupMirage } from 'ember-cli-mirage/test-support';
...
-
 module('Acceptance | clients | overview', function (hooks) {
   setupApplicationTest(hooks);
   setupMirage(hooks);
 
   hooks.beforeEach(async function () {
     sinon.replace(timestamp, 'now', sinon.fake.returns(STATIC_NOW));
     clientsHandler(this.server);
+    // stub secrets sync being activated
+    this.server.get('/sys/activation-flags', function () {
```
Comment (hashishaw) [inline ui/tests/acceptance/clients/counts/overview-test.js:30]: This wasn't being respected when it lived inside the test so I moved it here

File: ui/app/components/clients/activity.ts [hunk 4]
```diff
@@ -99,9 +89,14 @@ export default class ClientsActivityComponent extends Component<Args> {
   }
 
   // (object) top level TOTAL client counts for given date range
-  get totalUsageCounts() {
-    const { namespace, activity } = this.args;
-    return namespace ? this.filteredActivity : activity.total;
+  get totalUsageCounts(): TotalClients {
+    const { namespace, activity, mountPath } = this.args;
+    // only do this if we have a mountPath filter.
+    // namespace is filtered on API layer
```
Comment (hellobontempo) [inline ui/app/components/clients/activity.ts:95]: great comments, thank you!

File: ui/app/components/clients/page/counts.ts [hunk 5]
```diff
@@ -141,17 +141,17 @@ export default class ClientsCountsPageComponent extends Component<Args> {
   }
 
   get activityForNamespace() {
+    // TODO: always return activity for namespace. If no namespace selected, use the current
```
Comment (hellobontempo) [inline ui/app/components/clients/page/counts.ts:144]: Is this a TODO for this PR or for a follow-on?

**Review by hellobontempo — APPROVED** [review]
Well done! 🤩  Great comments explaining the util function, it makes the iterating much easier to follow. Just a question about a `todo` comment left behind.

File: ui/app/components/clients/page/counts.ts (hunk 5 above)
Comment (hashishaw) [reply ui/app/components/clients/page/counts.ts:144]: Follow-on, this will be addressed with the filter bar work

**Pull request merged**